### ISSUE TITLE
Automate Lazy sync  - immediate

### ISSUE THREAD

**Issue by preethit** [PulpQE/pulp-smash#71]
Documentation
https://github.com/pulp/pulp/blob/master/docs/dev-guide/design/deferred-download.rst

Option to use on with repo create

Download Policy
  --download-policy - content downloading policy (immediate | background |
                      on_demand)
Case 1
1. Create a repo with immediate download policy
pulp-admin rpm repo create --repo-id zoo --feed https://repos.fedorapeople.org/repos/pulp/pulp/demo_repos/zoo/  --download-policy immediate
2. Run repo Sync
3. The repo is synced and content is saved (check /var/lib/pulp/content )

**Comment by omaciel** [comment]
@preethit can you flush out what the test cases should be? Provide examples of what the expected behavior should be when using each of the allowed policy types? What about using a value other than the ones allowed?

**Comment by preethit** [comment]
I just started this  as I started playing with the feature. Will flush out as I get familiar and add more scenarios both positive & negative.

**Comment by bmbouter** [comment]
Consider splitting these into multiple issues so they could be done in parallel, reviewed, and merged independently of each other. I'm OK also not splitting them.

**Comment by omaciel** [comment]
@bmbouter :100:

**Comment by Ichimonji10** [comment]
We have tests for this in [`pulp_smash.tests.rpm.api_v2.test_download_policies`](http://pulp-smash.readthedocs.io/en/latest/api/pulp_smash.tests.rpm.api_v2.test_download_policies.html).
